src/golang/lib/models/shared/workflow.go

```diff
@@ -0,0 +1,47 @@
+package shared
+
+import (
+	"database/sql/driver"
+
+	"github.com/aqueducthq/aqueduct/lib/models/utils"
+	"github.com/google/uuid"
+)
+
+// `NotificationSettings` maps IntegrationID to NotificationLevel
+type NotificationSettings map[uuid.UUID]NotificationLevel
+
+func (s *NotificationSettings) Value() (driver.Value, error) {
+	return utils.ValueJSONB(*s)
+}
+
+func (s *NotificationSettings) Scan(value interface{}) error {
+	return utils.ScanJSONB(value, s)
+}
+
+type NullNotificationSettings struct {
+	NotificationSettings
+	IsNull bool
+}
+
+func (n *NullNotificationSettings) Value() (driver.Value, error) {
```

Comment:
@saurav-c this NullX pattern is something I mentioned the other day in slack, that I feel we can leverage template to do since `X` can be a generic type

Reply:
Got it, yeah lets create a separate task for that!